src/backend/metal/src/map.rs

```diff
@@ -440,77 +479,78 @@ pub fn map_base_type(ty: MTLDataType) -> shade::BaseType {
-        MTLTextureType::D1Array       => shade::TextureType::D1(Array),
-        MTLTextureType::D2            => shade::TextureType::D2(NoArray, NoMultiSample),
-        MTLTextureType::D2Array       => shade::TextureType::D2(Array, NoMultiSample),
+        MTLTextureType::D1 => shade::TextureType::D1(NoArray),
+        MTLTextureType::D1Array => shade::TextureType::D1(Array),
+        MTLTextureType::D2 => shade::TextureType::D2(NoArray, NoMultiSample),
+        MTLTextureType::D2Array => shade::TextureType::D2(Array, NoMultiSample),
         MTLTextureType::D2Multisample => shade::TextureType::D2(NoArray, MultiSample),
-        MTLTextureType::D3            => shade::TextureType::D3,
-        MTLTextureType::Cube          => shade::TextureType::Cube(NoArray),
-        MTLTextureType::CubeArray     => shade::TextureType::Cube(Array),
+        MTLTextureType::D3 => shade::TextureType::D3,
+        MTLTextureType::Cube => shade::TextureType::Cube(NoArray),
+        MTLTextureType::CubeArray => shade::TextureType::Cube(Array),
     }
 }
 
 pub fn map_texture_bind(bind: Bind) -> MTLTextureUsage {
     let mut flags = MTLTextureUsageUnknown;
 
-    if bind.contains(factory::RENDER_TARGET) || bind.contains(factory::DEPTH_STENCIL) {
+    if bind.contains(memory::RENDER_TARGET) || bind.contains(memory::DEPTH_STENCIL) {
         flags = flags | MTLTextureUsageRenderTarget;
     }
 
-    if bind.contains(factory::SHADER_RESOURCE) {
+    if bind.contains(memory::SHADER_RESOURCE) {
         flags = flags | MTLTextureUsageShaderRead;
     }
 
-    if bind.contains(factory::UNORDERED_ACCESS) {
+    if bind.contains(memory::UNORDERED_ACCESS) {
         flags = flags | MTLTextureUsageShaderWrite;
     }
 
     flags
 }
 
-pub fn map_access(access: MapAccess) -> MTLResourceOptions {
+pub fn map_access(access: memory::Access) -> MTLResourceOptions {
     match access {
-        MapAccess::Readable => MTLResourceCPUCacheModeDefaultCache,
-        MapAccess::Writable => MTLResourceCPUCacheModeWriteCombined,
-        MapAccess::RW => MTLResourceCPUCacheModeDefaultCache,
+        memory::READ => MTLResourceCPUCacheModeDefaultCache,
+        memory::WRITE => MTLResourceCPUCacheModeWriteCombined,
+        memory::RW => MTLResourceCPUCacheModeDefaultCache,
+        _ => unreachable!(),
     }
 }
 
 pub fn map_texture_usage(usage: Usage) -> (MTLResourceOptions, MTLStorageMode) {
     match usage {
         Usage::GpuOnly => (MTLResourceStorageModePrivate, MTLStorageMode::Private),
-
-        Usage::Const   => (MTLResourceStorageModePrivate, MTLStorageMode::Managed),
+        Usage::Immutable => (MTLResourceStorageModePrivate, MTLStorageMode::Managed),
         Usage::Dynamic => (MTLResourceCPUCacheModeDefaultCache, MTLStorageMode::Managed),
         Usage::CpuOnly(access) => (map_access(access), MTLStorageMode::Managed),
+        Usage::Persistent(access) => (map_access(access), MTLStorageMode::Managed),
```

Comment:
IIRC it might be just copy-pasted and I haven't gotten around to changing it yet, but fixing it shouldn't be part of this PR though